nilearn/_utils/estimator_checks.py

```diff
@@ -369,7 +369,7 @@ def return_expected_failed_checks(
         if (
             not IS_SKLEARN_1_6_1_on_py_lt_3_13
             and SKLEARN_MINOR >= 5
-            and scipy_version != "1.8.0"
+            and scipy_version != "1.9.0"
```

Comment:
Should `scipy_version` be compared to `1.9.0` or this condition be removed? 

It seemed to me like, the below check should not be done with scipy `1.8.0` but for greater versions it should stay.

Reply:
ideally yes we should identify the version where the behavior starts to change and have an <= or >=, but I am being lazy: I have not checked all the scipy versions to know when this happens: I think we are at least 10 minor versions behind the latest scipy